benchopt/cli/tests/test_cli.py

```diff
@@ -336,7 +336,7 @@ def test_caching(self, n_rep):
             # Check that the computation caching is working properly.
             run_cmd = (
                 f"{bench.benchmark_dir} -d test-dataset -s test-solver "
-                f"-n 1 -r {n_rep} --no-plot"
+                f"-n 1 -r {n_rep} --no-plot --seed 0"
```

Comment:
This might be an issue: there are some cases where there is no randomness in the benchmark, how to ensure that they are correcly cached without needing the user to specify it.

Maybe the seed should only impact the caching of classes defined where `get_seed` is called.